### PR TITLE
pkg/longpath: deprecate Prefix const, and use early returns in AddPrefix

### PR DESCRIPTION
**Pull request by thaJeztah**
- relates to https://github.com/moby/moby/pull/40648

### pkg/longpath: deprecate Prefix const

This const was exported because it was in use by pkg/symlink. This
dependency was removed in a48c6e30057bed6092ffe82cd1976c7ae0ec227d,
after which this const was only used internally.

This patch deprecates the const and introduces a non-exported const
to use.

There are no known external consumers of this const, so we may skip
deprecating it.

### pkg/longpath: AddPrefix: use early returns

**- What I did**

**- Description for the changelog**

```markdown changelog
deprecate pkg/longpath.Prefix const
```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
@laurazard @vvoland ptal 🤗

File: pkg/longpath/longpath.go [hunk 1]
```diff
@@ -12,20 +12,22 @@ import (
 )
 
 // Prefix is the longpath prefix for Windows file paths.
-const Prefix = `\\?\`
```
Comment (thaJeztah) [inline pkg/longpath/longpath.go:15]: Arf.. I lost the actual `deprecated` comment 😂  let me fix

**Comment by laurazard** [comment]
> There are no known external consumers of this const, so we may skip
deprecating it.

But it looks like the patch deprecates it? I was a little confused 😅

**Review by vvoland — APPROVED** [review]

**Comment by thaJeztah** [comment]
> But it looks like the patch deprecates it? I was a little confused 😅

😅 sorry poor wording there; I guess I wanted to write; but we could also decide to skip doing a deprecation first.

**Pull request merged**